libtiledbsoma/src/soma/soma_array.cc

```diff
@@ -1882,6 +1882,189 @@ void SOMAArray::_set_soma_joinid_shape_helper(
     schema_evolution.array_evolve(uri_);
 }
 
+std::pair<bool, std::string> SOMAArray::can_upgrade_domain(
```

Comment:
Did we add a `using StatusAndReason = std::pair<bool, std::string>` alias in a previous PR? I thought we did but now can't find it.

Reply:
oh yes -- indeed, let me check 👀

Reply:
I was puzzled too! Turns out the reason neither of us could find this is it was done in Python not C++ :D 
https://github.com/single-cell-data/TileDB-SOMA/blob/62dd129515c8e6928d910a2d20f6e0ac431c9fe7/apis/python/src/tiledbsoma/_types.py#L82-L84

It's worth adapting here too though.

Reply:
(added)